lzy/lzy-service/src/main/java/ai/lzy/service/operations/OperationRunnersFactory.java

```diff
@@ -0,0 +1,308 @@
+                                   GraphDao graphDao, ExecutionOperationsDao execOpsDao,
+                                   OperationsExecutor executor, KafkaAdminClient kafkaAdminClient,
+                                   KafkaLogsListeners kafkaLogsListeners, AllocatorBlockingStub allocClient,
+                                   BeanFactory.S3SinkClient s3SinkClient,
+                                   LongRunningServiceBlockingStub allocOpClient, SubjectServiceGrpcClient subjectClient,
+                                   AccessBindingServiceGrpcClient abClient, GraphExecutorBlockingStub graphsClient,
+                                   LzyChannelManagerPrivateBlockingStub channelManagerClient,
+                                   LongRunningServiceBlockingStub channelManagerOpClient,
+                                   RenewableJwt internalUserCredentials,
+                                   StorageClientFactory storageClientFactory, LzyServiceMetrics metrics,
+                                   VmPoolServiceBlockingStub vmPoolClient, KafkaConfig kafkaConfig,
+                                   IdGenerator idGenerator)
+    {
+        this.storage = storage;
+        this.wfDao = wfDao;
+        this.execDao = execDao;
+        this.opDao = opDao;
+        this.graphDao = graphDao;
+        this.execOpsDao = execOpsDao;
+        this.executor = executor;
+        this.kafkaAdminClient = kafkaAdminClient;
+        this.kafkaLogsListeners = kafkaLogsListeners;
+        this.allocClient = allocClient;
+        this.s3SinkClient = s3SinkClient;
+        this.allocOpClient = allocOpClient;
+        this.subjectClient = subjectClient;
+        this.abClient = abClient;
+        this.graphsClient = graphsClient;
+        this.channelManagerClient = channelManagerClient;
+        this.channelManagerOpClient = channelManagerOpClient;
+        this.internalUserCredentials = internalUserCredentials;
+        this.storageClientFactory = storageClientFactory;
+        this.metrics = metrics;
+        this.vmPoolClient = vmPoolClient;
+        this.kafkaConfig = kafkaConfig;
+        this.idGenerator = idGenerator;
+    }
+
+    public StartExecution createStartExecOpRunner(String opId, String opDesc, @Nullable String idempotencyKey,
+                                                  String userId, String wfName, String execId,
+                                                  Duration allocateVmCacheTimeout, PortalVmSpec portalVmSpec)
+        throws Exception
+    {
+        LMST.StorageConfig storageConfig = withRetries(LOG, () -> execDao.getStorageConfig(execId, null));
+
+        var allocatorSessionSpec = new AllocatorSessionSpec(userId, "session for exec with id=" + execId,
+            VmAllocatorApi.CachePolicy.newBuilder()
+                .setIdleTimeout(Durations.fromSeconds(allocateVmCacheTimeout.getSeconds()))
+                .build());
+
+        return StartExecution.builder()
```

Comment:
do we really need this builder? why not ctor?

Reply:
because of amount parameters